src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Dataflow/DiagnosticUtilities.cs

```diff
@@ -78,6 +78,9 @@ internal static bool TryGetRequiresAttribute(TypeSystemEntity member, string req
                 case PropertyPseudoDesc property:
                     decoded = property.GetDecodedCustomAttribute("System.Diagnostics.CodeAnalysis", requiresAttributeName);
                     break;
+                case EventPseudoDesc @event:
+                    decoded = @event.GetDecodedCustomAttribute("System.Diagnostics.CodeAnalysis", requiresAttributeName);
```

Comment:
Nit: Maybe create a const for the namespace name.

Reply:
I think I will move the constant strings out of the DiagnosticUtilities to the ILLink.Shared project, that way they can be used by everyone (there are files in the ILLink.Shared project that searches by string for attributes).

Reply:
Good idea - please make sure you backport these changes to dotnet/linker.